aten/src/ATen/native/FractionalMaxPool3d.cpp

```diff
@@ -168,9 +168,15 @@ void fractional_max_pool3d_out_cpu_template(
   int64_t widthDim = 3;
 
   int64_t ndims = input_.ndimension();
-  TORCH_CHECK(input_.numel() != 0 && (ndims == 4 || ndims == 5),
-    "fractional_max_pool3d_out(): non-empty 4D or 5D (batch mode) tensor ",
-    " expected for input, but got: ", ndims);
+  TORCH_CHECK(ndims == 4 || ndims == 5,
+              "fractional_max_pool3d_out(): Expected 4D or 5D tensor with optional 0-dim batch size, but got: ",
```

Comment:
nit: I think since this check is only concerned with the number of dims, "fractional_max_pool2d(): Expected 3D or 4D tensor, but got: " is better